### PR TITLE
fix(component): lit portal not re-rendering in inline links case

### PR DESCRIPTION
**Pull request by fundon**

### CODE REVIEW

**Comment by fundon** [comment]
* **#9354** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9354?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#9321** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9321?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9321?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/toeverything?ref=merge-queue-instructions-in-stack-comment&prId=9321).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by pengx17 — COMMENTED** [review]
please add test case

**Comment by fundon** [comment]
> please add test case

* inline links test cas:
https://github.com/toeverything/AFFiNE/blob/canary/tests/affine-local/e2e/links.spec.ts#L609

* i will add pdf embed view test case in pdf PR

**Review by EYHN — APPROVED** [review]

File: packages/frontend/core/src/components/blocksuite/block-suite-editor/lit-adaper.tsx [hunk 1]
```diff
@@ -194,7 +194,7 @@ const usePatchSpecs = (shared: boolean, mode: DocMode) => {
       () => (
         <>
           {portals.map(p => (
-            <Fragment key={p.id}>{p.portal}</Fragment>
+            <Fragment key={p.portal.key}>{p.portal}</Fragment>
```
Comment (pengx17) [inline packages/frontend/core/src/components/blocksuite/block-suite-editor/lit-adaper.tsx:197]: I think this change is unnecessary

Comment (fundon) [reply packages/frontend/core/src/components/blocksuite/block-suite-editor/lit-adaper.tsx:197]: `portal.id` has been updated and cannot be used anymore

Comment (fundon) [reply packages/frontend/core/src/components/blocksuite/block-suite-editor/lit-adaper.tsx:197]: `portal.id` is mainly used when querying here:

```ts
if (name === 'disconnectedCallback') {
    setPortals(portals => portals.filter(p => p.id !== id));
    return;
}
```

Comment (pengx17) [reply packages/frontend/core/src/components/blocksuite/block-suite-editor/lit-adaper.tsx:197]: The ID/key etc are very confusing. It seems unnecessary complicated. I have updated the code in the upstack #9354. Can you please take a look?

**Review by pengx17 — APPROVED** [review]

**Pull request merged**